mingw-64-brotli/PKGBUILD

```diff
@@ -0,0 +1,109 @@
+# Maintainer: J. Peter Mugaas <jpmugaas@suddenlink.net>
+
+_realname=brotli
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}"
+         "${MINGW_PACKAGE_PREFIX}-python-${_realname}"
+         "${MINGW_PACKAGE_PREFIX}-brotli-testdata")
+pkgver=1.0.9
+pkgrel=4
+pkgdesc="Brotli compression library (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
+url='https://github.com/google/brotli'
+makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-ninja"
+             "${MINGW_PACKAGE_PREFIX}-python")
+license=('MIT')
+source=("${_realname}-${pkgver}.tar.gz"::"https://github.com/google/brotli/archive/v${pkgver}.tar.gz"
+        '09b0992b6acb7faa6fd3b23f9bc036ea117230fc.patch')
+sha256sums=('f9e8d81d0405ba66d181529af42a3354f838c939095ff99930da6aa9cdf6fe46'
+            '0b1c8045b4fa745e620b5c1a75377f9cda839fbb07c90bf3358b583866f063ee')
+
+prepare() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  patch -p1 -i "${srcdir}/09b0992b6acb7faa6fd3b23f9bc036ea117230fc.patch"
+}
+
+build() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
+    ${MINGW_PREFIX}/bin/python setup.py build
+
+  [[ -d "${srcdir}"/build-${CARCH} ]] && rm -rf "${srcdir}"/build-${CARCH}
+  mkdir -p "${srcdir}"/build-${CARCH} && cd "${srcdir}"/build-${CARCH}
+
+  declare -a extra_config
+  if check_option "debug" "n"; then
+    extra_config+=("-DCMAKE_BUILD_TYPE=Release")
+  else
+    extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    ${MINGW_PREFIX}/bin/cmake \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+      "${extra_config[@]}" \
+      -DBUILD_STATIC_LIBS=ON \
+      -DBUILD_SHARED_LIBS=OFF \
```

Comment:
Apologies, I guess I didn't test thoroughly enough.

> Removing the static libraries hasn't worked; the artifacts still contain .dll and .dll.a files. Why do you build with
> ```-DBUILD_{SHARED,STATIC}_LIBS=ON```


That was the part I thought I had changed here. I'll investigate this further.

Reply:
Found the issue. Brotli actually doesn't yet support `-DBUILD_SHARED_LIBS=OFF`. There is a PR that's been opened to address that: https://github.com/google/brotli/pull/655

There are targets I can use for just the static libraries, but the `install` target will always build the shared libraries.

Reply:
You can include that PR in the package as a patch, e.g. https://github.com/google/brotli/pull/655.diff

Reply:
Thanks. Ended up doing that.

(There was one oddity: the packaged source code (zip and tar.gz) on Github does not contain the `c/fuzz/` directory, but the tagged commit on GitHub does. I just deleted that section from the patch manually.)